bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -0,0 +1,508 @@
+import com.adobe.acs.commons.redirects.models.RedirectMatch;
+import com.adobe.acs.commons.redirects.models.RedirectRule;
+import com.adobe.granite.jmx.annotation.AnnotatedStandardMBean;
+import com.day.cq.replication.ReplicationAction;
+import com.day.cq.wcm.api.WCMMode;
+import org.apache.commons.lang.StringUtils;
+import org.apache.sling.api.SlingConstants;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.request.RequestPathInfo;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.apache.sling.api.resource.observation.ResourceChange;
+import org.apache.sling.api.resource.observation.ResourceChangeListener;
+import org.apache.sling.engine.EngineConstants;
+import org.osgi.service.component.annotations.*;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.management.NotCompliantMBeanException;
+import javax.management.openmbean.*;
+import javax.servlet.*;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.apache.sling.engine.EngineConstants.SLING_FILTER_SCOPE;
+import static org.osgi.framework.Constants.SERVICE_DESCRIPTION;
+import static org.osgi.framework.Constants.SERVICE_RANKING;
+
+/**
+ * A request filter that implements support for virtual redirects.
+ */
+@Component(service = {Filter.class, RedirectFilterMBean.class, EventHandler.class}, property = {
+        SERVICE_DESCRIPTION + "=A request filter implementing support for virtual redirects",
+        SLING_FILTER_SCOPE + "=" + EngineConstants.FILTER_SCOPE_REQUEST,
+        SERVICE_RANKING + ":Integer=10000",
+        "jmx.objectname=" + "com.adobe.acs.commons:type=Redirect Manager",
+        EventConstants.EVENT_TOPIC + "=" + ReplicationAction.EVENT_TOPIC,
+        ResourceChangeListener.CHANGES + "={ADDED, REMOVED, CHANGED}"
+
+})
+@Designate(ocd = RedirectFilter.Configuration.class)
+public class RedirectFilter extends AnnotatedStandardMBean
+        implements Filter, EventHandler, ResourceChangeListener, RedirectFilterMBean {
+
+    public static final String DEFAULT_STORAGE_PATH = "/var/acs-commons/redirects";

```

Comment:
Due to the limitations with /var in Content-Packages for AEM Cloud I would rather move that to /conf (https://experienceleague.adobe.com/docs/experience-manager-learn/cloud-service/debugging/debugging-aem-as-a-cloud-service/build-and-deployment.html?lang=en#deploy-to)

Reply:
@kwin thanks for the speedy feedback. 
+1 to move it to /conf.
 Which will be better, /conf/redirects or /conf/acs-commons/redirects ? 
I'm inclined towards /conf/acs-commons as we can re-use it for other components.

Reply:
@kwin thanks for the speedy feedback. 
+1 to move it to /conf.
 Which will be better, /conf/redirects or /conf/acs-commons/redirects ? 
I'm inclined towards /conf/acs-commons as we can re-use it for other components.

Reply:
We have `/conf/acs-commons/reports` in the 5.0.0 releae branch (this too had to get moved out of `/var`) so i vote for `/var/acs-commons/redirects` to keep `/conf` clean.

This is probably too much work (and might not be as good of an idea even) - but `/conf` is?/was? for CAConfig which is applied TO content trees. So following that paradigm Redirect configs could be created under `/conf/tenant-a` and then applied to content paths (ex. `/content/tenant-a` on a more granular level - vs having a global dashboard.
Or it could be put under `/conf/global` ..

anyhow - I digress, I'm not very passionate about the above since it doesn't seem like there is much rhyme or reason to what/how stuff is created/used in `/conf` anymore :)

https://github.com/Adobe-Consulting-Services/acs-aem-commons/tree/acs-aem-commons-5.0.0/ui.content/src/main/content/jcr_root/conf/acs-commons/reports

Reply:
@davidjgonzalez Can you clarify
> so i vote for /var/acs-commons/redirects to keep /conf clean.

I am confused, so you want that to stay in /var (with the known issues with Cloud Manager Deployment) or was that a typo?

Reply:
Sorry @kwin  - i had critical typo :)

I vote for `/conf/acs-commons/redirects` to keep `/conf` clean and align with the move of `/conf/acs-commons/reports`

Reply:
I updated the code to store redirects /conf/acs-commons/redirects